hank-core/src/main/java/com/liveramp/hank/zookeeper/ZooKeeperPlus.java

```diff
@@ -375,12 +375,26 @@ public void ensureCreated(String path, byte[] value) throws InterruptedException
     }
 
     public void ensureCreated(String path, byte[] value, CreateMode createMode) throws InterruptedException, KeeperException {
-      if (exists(path, false) == null) {
+      if (!path.isEmpty() && exists(path, false) == null) {
+        ensureCreated(getParent(path), null, createMode);
         create(path, value, DEFAULT_ACL, createMode);
         NodeCreationBarrier.block(ZooKeeperPlus.this, path);
       }
     }
 
+    private boolean isRootOrEmpty(String path){
+      return path.isEmpty() || path.equals("/");
+    }
+
+    // TODO I don't want to create a File, and idk what other API I can use for htis
+    private String getParent(String path){
```

Comment:
Think you can use `File`. Constructing an instance doesn't actually create anything -- only when you call `createNewFile`, etc. does it actually touch the fs.